### ISSUE TITLE
[RFE] Add TF script for devs and customers to do local testing

### ISSUE THREAD

**Issue by a-blender** [rancher/terraform-provider-rancher2#1156]
### Is your feature request related to a problem? Please describe.

Devs and customers do _a lot_ of local testing with terraform. There's two ways to test locally: 1) configure Terraform to use a local binary that you've generated, or 2) use an `in-dev` version of the provider with `dev-overrides` [relevant article](https://nicovibert.com/2021/03/29/dev-overrides-terraform/), thank you @git-ival.

### Describe the solution you'd like

A script to automate the process of generating and running a local binary.

**Comment by a-blender** [comment]
# QA Test Template

Terraform RC: v3.0.2-rc2
  
# Problem

  
Devs and customers do _a lot_ of local testing with Terraform. There's two ways to test locally: 1) configure Terraform to use a local binary that you've generated, or 2) use an `in-dev` version of the provider with `dev-overrides` [relevant article](https://nicovibert.com/2021/03/29/dev-overrides-terraform/) and it's all painstakingly manual. Manual testing also requires knowledge on where the local binary gets generated, how to configure TF to reference it, etc.

# Solution

  
Add a bash script to automate building a binary from your local repo. Include script options to run `terraform init` and/or `terraform apply` in the pipeline.

# Testing


## Engineering Testing

### Manual Testing


* Clone new rancher/terraform-provider-rancher2 repo
* Write tf config that includes

```
rraform {
 required_providers {
     rancher2 = {
       source = "terraform.local/local/rancher2"
       version = "1.0.0"
     }
   }
 }
```

Run commands

```
./scripts/build_local_binary.sh
./scripts/build_local_binary.sh -i
./scripts/build_local_binary.sh -i -a
```

Verify binary is present at `LOCAL_BINARY_PATH` on fs. And that `terraform init` is run with option -i and both init and apply are run with option -a.

Basically, you should be able to make changes to your local repo, run this script with apply option which will build, input config, plan, and execute a build infrastructure plan within 2 seconds. Make changes, run script to test. Make changes, run script to test. Automation forever, baby ;)
 
### Automated Testing

 
N/A

## QA Testing Considerations

  
### Regressions Considerations


N/A

**Comment by slickwarren** [comment]
To Be Tested On:
Windows
Mac
Linux

**Comment by Josh-Diamond** [comment]
Tested on Mac + Linux - ✅  functional + no issues seen 🎉 
---

| Scenario | Test Case (OS) | Result |
|---|---|---|
| 1. | MacOS | ✅ |
| 2. | (Amazon) Linux | ✅ |

**Comment by Josh-Diamond** [comment]
Ticket to be closed upon successful Windows test case - _pending_

**Comment by thaneunsoo** [comment]
Tested on windows and it seems to be working as expected as well.
![image](https://github.com/rancher/terraform-provider-rancher2/assets/108360907/8aff6bdc-037c-402f-bde5-2c3c5ca214f8)

Closing this ticket since linux mac and windows has been verified.

**Comment by a-blender** [comment]
@Josh-Diamond is doing some re-testing on this to fully verify the Mac + Linux cases.

**Comment by a-blender** [comment]
@thaneunsoo Based on my [test plan](https://github.com/rancher/terraform-provider-rancher2/issues/1156#issuecomment-1674196813), this https://github.com/rancher/terraform-provider-rancher2/issues/1156#issuecomment-1726195731 is the wrong test case. `setup-provider-windows.ps1` is for a [different issue](https://github.com/rancher/terraform-provider-rancher2/issues/1083) - we are only testing `build_local_binary.sh` here. Can you move that test result to the relevant issue please? Thank you.

**Comment by Josh-Diamond** [comment]
Ticket #1156 - Test Results - ✅ 
---

| Scenario | Test Case | Result |
|---|---|---|
| 1. | MacOS - `build_local_binary.sh` | ✅ |
| 2. | MacOS - `build_local_binary.sh -i` | ✅ |
| 3. | MacOS - `build_local_binary.sh -i -a` | ✅ |
| 4. | (Amazon) Linux - `build_local_binary.sh` | ✅ |
| 5. | (Amazon) Linux - `build_local_binary.sh -i` | ✅ |
| 6. | (Amazon) Linux - `build_local_binary.sh -i -a` | ✅ |

script working as expected on both MacOS + (Amazon) Linux; all script options were tested - 🚀 

script is n/a to WIndows, so closing out the issue now.
